### ISSUE TITLE
Should pull packages from bioconductor as well

### ISSUE THREAD

**Issue by hadley** [hadley/crantastic#9]
http://bioconductor.org/docs/install/

**Comment by hadley** [comment]
Similarly, should we consider pulling in packages in development from (r-forge)[http://r-forge.r-project.org] as well?

We might need some badging on the packages to indicate their source, and to somehow deal with packages that are present in multiple sources.  For example, hexbin is in cran and bioconductor, and many of the packages in r-forge will have releases on cran.

**Comment by bmaland** [comment]
Yes, this will require some thought. I believe it might be harder to pull stuff from r-forge, or do they have some kind of API as well?

**Comment by hadley** [comment]
R-forge behaves like a normal R repository, so it should be possible to pull down data in the same way as from cran.

**Comment by bmaland** [comment]
Ah, for some reason I thought R-forge differed from the others.

**Comment by hadley** [comment]
Ideally, we should just have a top level list of all repositories we want to pull packages in from.  Then if we discover a new source we can just add it to the list.

**Comment by bmaland** [comment]
Bioconductor has a more complex structure than CRAN, with several sub-repositories. From their install script:

```
## Bioconductor CRAN-style repositories.
## The software repo (bioc) _must_ be the first element.
repos <- c(
    "2.4/bioc",
    "2.4/data/annotation",
    "2.4/data/experiment",
    "2.4/extra"
)
repos <- paste("http://bioconductor.org/packages", repos, sep="/")
c(repos, "http://brainarray.mbni.med.umich.edu/bioc", "http://cran.fhcrc.org")
```

Do we need all of these or is the main one enough? (http://bioconductor.org/packages/2.4/bioc/)

Since they keep version numbers in the URLs these has to be updated when new versions of Bioconductor are released.

**Comment by bmaland** [comment]
There are differences between the DESCRIPTION files in CRAN and Bioconductor (probably r-forge as well), so special rules has to be added for each. Bioconductor includes a list of views right in the DESCRIPTION file, which is quite useful for our purposes. E.g. "biocViews: DNACopyNumber, DataImport, Genetics"

**Comment by hadley** [comment]
I think the main bioconductor repository will be sufficient, and yes, it's a pain that we'll have to keep changing the url.  Can we treat biocViews in a similar way to task views?

**Comment by egbutter** [comment]
hi guys: mind pointing me to the code that yanks from cran?  i will try to yank in bioconductor ... would be useful for me to keep track of which packages i am using.  cheers,

**Comment by bmaland** [comment]
The relevant files are https://github.com/hadley/crantastic/blob/master/lib/cran.rb and https://github.com/hadley/crantastic/blob/master/lib/crantastic.rb , they are used from the rake task scripts locaed in lib/tasks
